src/DocumentStore.zig

```diff
@@ -1335,6 +1335,14 @@ pub fn uriFromImportStr(self: *DocumentStore, allocator: std.mem.Allocator, hand
                     return try URI.fromPath(allocator, pkg.path);
                 }
             }
+        } else if (isBuildFile(handle.uri)) {
+            const build_file_entry = self.getOrLoadBuildFile(handle.uri) orelse return null;
+            const config = build_file_entry.value_ptr.config orelse return null;
```

Comment:
Similar to the what I've written above, this code has a potential race condition since pointers to build files are not stable and may be invalidated immediately after `getOrLoadBuildFile` has returned unless you are holding a lock to the document store. I will need to do some refactoring to make it easier to write this in a safe way.

The code that I've written above has also a race condition because we are reading from `build_file.config` without holding a lock. This case is much easier to fix than your code (just re-aquire a lock after `getBuildFile`).

You can let me fix this as follow up.

Reply:
Ah, I should've probably used `getBuildFile`, as this matters only when it is loaded and being edited.

Reply:
#1575 should solve this.